src/coreclr/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackFrame.NativeAot.cs

```diff
@@ -28,6 +30,39 @@ public partial class StackFrame
         /// </summary>
         private bool _needFileInfo;
 
+        /// <summary>
+        /// Will be true if we attempted to retrieve the associated MethodBase but couldn't.
+        /// </summary>
+        private bool _noMethodBaseAvailable;
+
+        /// <summary>
+        /// Returns the method the frame is executing
+        /// </summary>
+        [RequiresUnreferencedCode("Metadata for the method might be incomplete or removed")]
+        public virtual MethodBase? GetMethod()
+        {
+            TryInitializeMethodBase();
+            return _method;
+        }
+
+        private bool TryInitializeMethodBase()
+        {
+            if (_noMethodBaseAvailable || _ipAddress == IntPtr.Zero)
+                return false;
+
+            if (_method != null)
+                return true;
+
+            IntPtr methodStartAddress = RuntimeImports.RhFindMethodStartAddress(_ipAddress);
```

Comment:
This can be just `_ipAddress - _nativeOffset`. (We can assert that it is same value as what `RhFindMethodStartAddress` returns.)